perfkitbenchmarker/pkb.py

```diff
@@ -428,13 +436,46 @@ def RunBenchmarks(publish=True):
       vm_util.RunThreaded(
           RunBenchmark, args, max_concurrent_threads=FLAGS.parallelism)
     else:
-      for run_args, kwargs in args:
-        RunBenchmark(*run_args, **kwargs)
+      while args:
+        run_args, _ = args.popleft()
+        benchmark_module, _, sequence_number, _, _, benchmark_uid = run_args
+        benchmark_name = benchmark_module.BENCHMARK_NAME
+        try:
+          RunBenchmark(*run_args)
+          benchmark_arg_succeeded_tuples.append((run_args, True))
+        except BaseException as e:
+          benchmark_arg_succeeded_tuples.append((run_args, False))
+          msg = 'Benchmark {0}/{1} {2} (UID: {3}) failed.'.format(
+              sequence_number, total_benchmarks, benchmark_name, benchmark_uid)
+          if (isinstance(e, KeyboardInterrupt) or
+              FLAGS.stop_after_benchmark_failure):
+            logging.error('%s Execution will not continue.', msg)
+            break
+          logging.error('%s Execution will continue.', msg)
+      benchmark_arg_succeeded_tuples.extend((a, False) for a, _ in args)
 
   finally:
     if collector.samples:
       collector.PublishSamples()
 
+    if benchmark_arg_succeeded_tuples:
```

Comment:
Is the output different from what you had before though?

I was thinking of something interleaved like:

```
index  benchmark    status
0      iperf        SUCCESS
1      netperf      SUCCESS
2      cassandra    SUCCESS
3      netperf      FAILED
4      speccpu2006  SUCCESS
5      speccpu2006  FAILED
6      iperf        SKIPPED
7      netperf      SKIPPED
```

But up to you. Code LGTM.